ui/widgets/src/aptoide/java/com/appcoins/wallet/ui/widgets/OpenGameFunction.kt

```diff
@@ -6,20 +6,28 @@ import android.content.Intent
 import android.content.pm.PackageManager
 import android.net.Uri
 import androidx.core.content.ContextCompat.startActivity
+import com.appcoins.wallet.core.analytics.analytics.legacy.GetAppAnalytics
+import okhttp3.internal.applyConnectionSpec
 
-fun openGame(gamePackage: String?, actionUrl: String?, context: Context) {
+fun openGame(
+  gamePackage: String?,
+  actionUrl: String?,
+  context: Context,
+  getAppAnalytics: GetAppAnalytics?
+) {
   try {
     val launchIntent: Intent? = gamePackage?.let {
       context.packageManager.getLaunchIntentForPackage(
         it
       )
     }
-    if (launchIntent != null)
+    if (launchIntent != null) {
+      getAppAnalytics?.sendGetAppAEvent(gamePackage, GetAppAnalytics.OPEN_ACTION)
       startActivity(context, launchIntent, null)
-    else
-      getGame(gamePackage, actionUrl, context)
+    } else
+      getGame(gamePackage, actionUrl, context, getAppAnalytics)
   } catch (e: Throwable) {
-    getGame(gamePackage, actionUrl, context)
+    getGame(gamePackage, actionUrl, context, getAppAnalytics)
```

Comment:
is it safe? you're trying to call the same function that you got an exception

Reply:
the exception here does't come from the getGame. the getGame is the fallback when the game is not installed.